CMakeLists.txt

```diff
@@ -1,19 +1,12 @@
 cmake_minimum_required(VERSION 3.5)
 
 project(pplib
-    VERSION 1.0.5
     LANGUAGES C)
+set(VERSION 1.0.5)
 
 set(CMAKE_C_STANDARD 99)
 
-if(MSVC)
-    add_compile_options(/W4 /WX)
-else(MSVC)
-    add_compile_options(
-        -Wall -Wextra
-        -Wunused -Wimplicit -Wreturn-type -Wdeclaration-after-statement -Wno-unknown-pragmas -Wmissing-prototypes -Wmissing-declarations -Wparentheses -Wswitch -Wtrigraphs -Wcast-qual -Wcast-align -Wwrite-strings -Wold-style-definition -Wpointer-arith
-    )
-endif(MSVC)
```

Comment:
I understand your argument here. I just cannot figure out a nice way to keep the list of desired flags for testing (as many warnings enabled as possible) in the repository, while at the same time still be able to run separate steps for configuring / making / testing. The suggested script does all at once.

Reply:
You don't have to use the CMake script above.  You can also store the flags in the `CFLAGS` environment variable and just configure and build as normal, i.e.
```console
$ export CFLAGS="-Wall -Wextra -Wunused -Wimplicit -Wreturn-type -Wdeclaration-after-statement -Wno-unknown-pragmas -Wmissing-prototypes -Wmissing-declarations -Wparentheses -Wswitch -Wtrigraphs -Wcast-qual -Wcast-align -Wwrite-strings -Wold-style-definition -Wpointer-arith"
$ mkdir build
$ cd build
$ cmake ..
$ make -j
```